### DISCUSSION TITLE
Colima & VSCode Devcontainer

### DISCUSSION THREAD

**Discussion by jefferyng33106** [abiosoft/colima#596] [Q&A]
Hello, community,

Hardware and OS:
- Apple silicon
- Latest macOS
- Colima from brew
- Docker, Docker Compae from brew

I am trying to run the vscode dev container with Colima, but I bump into a binding issue which is complaining about port 22 already in use. Furthermore, the IDE (or maybe it's the container) keep restarting and then failed eventually. 

Also, the wired part is that with Docker Desktop, everything is going fine. So I have no idea why it wouldn't work.

I wonder if there is anyone already integrate Colima with the VSCode dev container and can lend me some help.

Thank you 🙇

**Comment by abiosoft** [comment]
If you are trying to expose port 22, there may be a clash with the SSH server.

Can you share the instructions or steps to reproduce the issue?

Thanks.

**Comment by jefferyng33106** [comment]
Hello @abiosoft 

Yes, I saw my instructions in the docker-compose exposed port 22, but I've changed it to, e.g., `2022`.  After changing it, I could make it build until the last step. But the thing is that I wouldn't be able to "launch" the services. It kept doing something similar to crash loop backoff when VSCode tried to attach to the container (and maybe even for probing) 😿  But with the docker desktop, I managed to make it works appropriately without changing anything (including the SSH server). So I wonder if I can (or need) to change some custom settings.

If it's not clear or you still prefer to have the instructions, I will remove some sensitive data inside and get back to you asap. Thank you 🙏

**Reply by jefferyng33106** [reply]
Hello @abiosoft , any hints? Thank you

**Reply by abiosoft** [reply]
Can you share your devcontainer.json? Maybe remove the sensitive details.
